bitbots_ball_filter/bitbots_ball_filter/ball_filter.py

```diff
@@ -122,11 +129,50 @@ def ball_callback(self, msg: PoseWithCertaintyArray, optional_distance=True):
             ball_buffer.header = msg.header
             ball_buffer.point = ball.pose.pose.position
             try:
-                self.ball = self.tf_buffer.transform(ball_buffer, self.filter_frame, timeout=rclpy.duration.Duration(seconds=0.3))
+                self.ball = self.tf_buffer.transform(ball_buffer, self.filter_frame,
+                                                     timeout=rclpy.duration.Duration(seconds=0.3))
                 self.ball_header = msg.header
             except (tf2.ConnectivityException, tf2.LookupException, tf2.ExtrapolationException) as e:
                 self.get_logger().warning(str(e))
 
+    def select_closest_to_prediction_ball(self, msg: PoseWithCertaintyArray):
+        balls = [ball for ball in msg.poses if
+                 ball.confidence > self.min_ball_confidence]  # only look at balls who have at least some minimal certainty
```

Comment:
It is also used in the other function. But you are right that if they already get thresholded in the vision, it might not be necessary to do so again. I do not know whether the option of setting a higher threshold for the filter is useful. What do you think?

Reply:
I would use only one threshold as it is more straight forward while debugging and has no obvious downsides.